### DISCUSSION TITLE
laser

### DISCUSSION THREAD

**Discussion by 161975** [arkypita/LaserGRBL#1670] [Q&A]
Sorry to bother people , but Noob really needs help.
I have a Voidmicro  engraver with 30W laser(40x40mm). It all works fine..( it has propriety software and it's own  control card.  ( I don't like it's software controls and heard good things about GRBL)
But this card won't recognize GRBL

I purchased a GRBL compatible card on eBay
All connected and working well, but the laser won't turn on when it is running the program.
 The laser's  top blue led is very dimly lit and the incoming voltage is 12.68(red+/black-  eg the 2 outer wires) and 3.4v (red+ /yellow).
The laser blue LED on the card is brightly lit.
Plugged into 12VDC and NOT connected to a laptop the laser voltages remain the same.
The blue LED on the card lights when the laser is connected
I don't really understand why this problem...Am I missing something?
Many thanks for any help ..........

**Comment by StuartB4** [comment]
Do you have a picture of the card you bought from ebay and how you have it all wired up.

**Reply by 161975** [reply]
The card is listed as :  2 Axis GRBL 1.1 CNC Laser Control System Router/Laser Engraver Control Board Offline controller USB Port Controller Card
![20211208_092222-001](https://user-images.githubusercontent.com/33650026/145104262-b6f82d34-559c-4dcb-92cf-abe7ddadca3e.jpg)

![20211208_092139-001](https://user-images.githubusercontent.com/33650026/145104356-df62d29a-6e3c-4f9f-a030-8d992b19e55a.jpg)

As mentioned all  the motors work ok,

**Comment by StuartB4** [comment]
Do you have $32 set to 1 to enable laser mode in GRBL ?
Also set $30 to 255.
$31 should be 0.

**Reply by 161975** [reply]
I changed the settings as you suggested and double checked it.
 It makes no difference/ The laser is not firing. 
On the board the laser LED is bright blue, yet on the laser housing the blue LED is very dim.--> It used to be bright blue when it was working OK on the other board.
Just now, I switched boards and the laser is working fine, yet switching back to (GRBL board) it is no-go.
The motors are going through the motion OK

**Comment by 161975** [comment]
The wires soldered to the bottom for the laser are NOT connected .It says on the pic also "NC"
The laser wires connected are from the top of the board via a JST connector


     On Wednesday, December 8, 2021, 05:02:59 PM GMT+13, Stuart ***@***.***> wrote:  
 

Have you not noticed, you have the yellow on the middle pin on top of the board but on the outside pin on the bottom.

**Comment by 161975** [comment]
Attached 2 photos of the laser. It may help.....
![20211209_114151-001](https://user-images.githubusercontent.com/33650026/145304475-313878a5-cd5e-4cfc-bfe7-64a1716b2ba6.jpg)
![20211209_114700-001](https://user-images.githubusercontent.com/33650026/145304486-ffaaecf9-fa80-42a8-b580-776e0674e141.jpg)

**Comment by StuartB4** [comment]
How many pins are on the other plug where the red and yellow wires are?
![Untitled](https://user-images.githubusercontent.com/16980026/145326577-4977e0f5-d980-4835-9509-3737b719afa7.png)



There could be another issue. 
Some manufactures use inverted PWM signals.
The standard is Off=0 Volts, On full power=5 volts. 
It maybe that the Voidmicro controller board uses Off=5 volts and On=full power=0

You would need to check the voltage coming from the PWM and Ground pins on the main controller board
when the laser is told to switch on. 

When the laser is off it should be almost 0 volts, when it's on it should
vary between 0 (off) and 5 volts (full power) as the power goes up and down.

**Comment by 161975** [comment]
**How many pins are on the other plug where the red and yellow wires are?**  3 pins R B Y .All are connected to a  switch, which limits the  X Axis travel & stops (I presume) the laser. The switch is fed by 4 black wires which go the controller board
When (RUNNING) laser at 100% volt varies 0.1 to 2.3V between neg & yellow PWM.  and  Neg to Pos  varies 10 to 12,2V
When at idle PWM to Pos =12.4V  and  PWM to Neg=0.06V
What beats me, is that as Black (neg from board goes to Neg as well as  Pos of laser JST conn. (Not connected to laser)  Yet there is no connection between the 2 (according to my Fluke meter)
I hope it all makes sense to you. Picture attached
![20211209_165432-001](https://user-images.githubusercontent.com/33650026/145335566-6e85415a-b3a7-466f-bc87-d92cda745181.jpg)

**Comment by 161975** [comment]
Thanks StuartB4 for your help

**ANYONE WITH SUGGESTIONS??**
